pkg/runtime/runtime.go

```diff
@@ -368,6 +382,10 @@ func (m *MosnRuntime) initStates(factorys ...*runtime_state.Factory) error {
 			m.errInt(err, "create state component %s failed", name)
 			return err
 		}
+		//inject secret to component
+		if err = m.Injector.InjectSecretRef(config.SecretRef, config.Metadata); err != nil {
```

Comment:
As far as I remember, some users didn't configurate `metadata`, so the `config.Metadata` here might be `nil`. 
That's why we check nil in `initPubSubs`:
![image](https://user-images.githubusercontent.com/26001097/177718743-525e5ab0-57e9-4ff9-aef4-828d3a304093.png)
So, we should check nil here.
For example, we can modify the `InjectSecretRef` function to:
```
if config.Metadata, err = m.Injector.InjectSecretRef(config.SecretRef, config.Metadata); err != nil {
```
and check nil map in it:

```
func (i *DefaultInjector) InjectSecretRef(items []*ref.Item, metaData map[string]string) (map[string]string ,error) {
        if metaData==nil{
            metaData=make(map[string]string)
        }
	if len(items) == 0 {
		return metaData, nil
	}
//...
      
```

Reply:
@seeflood  reviewed